framework/decode/dx12_consumer_base.h

```diff
@@ -83,7 +83,7 @@ class Dx12ConsumerBase : public MetadataConsumerBase, public MarkerConsumerBase
                                                            UINT                                     SrcDepthPitch)
     {}
 
-    void SetCurrentBlockIndex(uint64_t block_index) { current_block_index_ = block_index; }
+    virtual void SetCurrentBlockIndex(uint64_t block_index) override { block_index_ = block_index; }
```

Comment:
A few lines below (unfortunately github does not allow me to put a comment there) you need to update `GetCurrentBlockIndex` (line 95) to return `block_index_` instead of `current_block_index_`.
Also delete lines 100 to 102 since variable `current_block_index_` is no longer used